packages/happn-db-provider-loki/index.js

```diff
@@ -628,9 +627,18 @@ module.exports = class LokiDataProvider extends commons.BaseDataProvider {
           this.logger.error('failed persisting operation data', appendFailure);
           return callback(appendFailure);
         }
-        if (this.operationCount < this.settings.snapshotRollOverThreshold) {
+        let currentSize = 0;
+        if (this.settings.fileSizeDifferenceLimit > 0) {
+          currentSize = this.#getFileSize(this.settings.filename);
+        }
+        if (
+          this.operationCount < this.settings.snapshotRollOverThreshold &&
+          (this.settings.fileSizeDifferenceLimit === 0 ||
```

Comment:
Rather use

if (  currentSize === 0 || currentSize - this.baselineFileSize < this.settings.fileSizeDifferenceLimit )